src/test/java/cz/muni/fi/pa165/TrainerFightDaoTest.java

```diff
@@ -0,0 +1,275 @@
+package cz.muni.fi.pa165;
+
+import cz.muni.fi.pa165.kodemon.KodemonApplicationContext;
+import cz.muni.fi.pa165.kodemon.dao.TrainerFightDao;
+import cz.muni.fi.pa165.kodemon.dao.TrainerDao;
+import cz.muni.fi.pa165.kodemon.dao.GymDao;
+import cz.muni.fi.pa165.kodemon.entity.TrainerFight;
+import cz.muni.fi.pa165.kodemon.entity.Trainer;
+import cz.muni.fi.pa165.kodemon.entity.Gym;
+import cz.muni.fi.pa165.kodemon.enums.PokemonType;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.domain.Example;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+/**
+ * Unit tests of {@link TrainerFightDao}
+ *
+ * @author Miso Romanek
+ */
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@ContextConfiguration(classes = KodemonApplicationContext.class)
+public class TrainerFightDaoTest extends AbstractTestNGSpringContextTests {
+
+    @Inject
+    private TrainerFightDao trainerFightDao;
+
+    @Inject
+    private TrainerDao trainerDao;
+
+    @Inject
+    private GymDao gymDao;
+
+    private TrainerFight trainerFight;
+    private Trainer trainer;
+    private Gym gym;
+
+    @BeforeMethod
+    void prepare() {
+        assertThat(trainerFight, is(notNullValue(TrainerFight.class)));
```

Comment:
asi si chcel napisat `trainerFightDao`, resp. `TrainerFightDao.class`
lebo trainerFight bude urcite `null`. Tak ja to teda upravim.